settings/security/third-party/saml.rst

```diff
@@ -19,81 +22,78 @@ Configure SAML in Zammad
+   
+   .. note::
+      - The above certificate is just an example, you should use a proper certificate from your IdP!
+      - Note that this is generally **not** the same cert used by TLS in the HTTPS connection to the IdP.
+
+   .. hint:: **ADFS users**
+      
+      Above certificate refers to the to the signing certificate.
+
+   In Keycloak, this certificate can be found by going to:
+      
+      - Keycloak Admin: Realm settings -> keys -> RSA -> "certifcate" popup
+      - Paste as a single line string with no line breaks.
+      - Do not set this option at all if using the "certificate fingerprint" for cert validation.
+
+IDP certificate fingerprint:
+   Optionally use the fingerprint instead of the certificate itself for cert validation.
+   Because sha1 is used for the fingerprint, this option is not preferred.
+   Do not set this option at all if using the "certificate" field above.
 
 **Name Identifier Format:**
-
-``urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress``
-
-
-Configured example:
-
-.. image:: /images/settings/security/third-party/saml/zammad_connect_saml_thirdparty.png
-   :alt: Zammad SAML Example
-
+   ``urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress``
 
 Configure the SAML IdP
 ----------------------
 
 If your IdP supports xml import or auto xml metadata retrieval, use this URL:
-``https://zammadFQDN/auth/saml/metadata``
+``https://{Zammad-FQDN}/auth/saml/metadata``
 
 For Keycloak IdP, save the contents of that metadata file as "metadata.xml" and then log into Keyclaok and import it as a new Client.
 
 Keycloak sends the "Username" in whatever format your users are stored in. The tested SAML instance used email address for the "username", but Zammad also needs to see an "email" attribute to correctly automap to existing users.
 In Keycloak, we can create a "Mapper" for the newly imported Zammad client.
 
-Goto -> Keycloak Admin -> Clients -> ``https://zammadFQDN/auth/saml/metadata`` -> Mappers -> Create
+Goto -> Keycloak Admin -> Clients -> ``https://{Zammad-FQDN}/auth/saml/metadata`` -> Mappers -> Create
 
 .. code-block:: html
 
    Name: ZammadEmail
-   Mapper Type: User Attribute
+   Mapper Type: User Property
    User Attribute: emailAddress
    SAML Attribute Name: email
    SAML Attribute NameFormat: Basic
```

Comment:
I've rephrased this entire section thusly:

![Screenshot_20200728_234748](https://user-images.githubusercontent.com/12194123/88689269-dd501280-d12c-11ea-9833-8b00d6dae1d5.png)

Is that a reasonable/accurate rewriting of this section?

#### Notes

1. In the latest version of Keycloak, "User Attribute" appears to have been renamed to "Property":

   ![Screenshot_20200728_233306](https://user-images.githubusercontent.com/12194123/88687372-c3adcb80-d12a-11ea-8f2d-c343fac7b6da.png)

2. I changed `User Attribute: emailAddress` to `Property: username` because just above this, the author talks about how a) his keycloak uses email addresses as usernames, and b) Zammad needs a valid email attribute mapping. Mapping usernames to emails seemed like the logical conclusion to me... but maybe I misunderstood something?

Reply:
Your rewrite looks good and reasonable!

It's a good thing to adapt to current naming conventions or keycloak in this regard.

Regarding 2b: I believe that this won't work. The reason for this is that Zammad expects a mail address and (so I expect) does lookup a mail address in it's database. This thus shouldn't work in my understanding.

Reply:
Totally get it. How is this phrasing?

![Screenshot_20200906_235701](https://user-images.githubusercontent.com/12194123/92329816-af9ba900-f09c-11ea-88cf-bd736f9b3f71.png)